### ISSUE TITLE
Notification won't dealloc unless

### ISSUE THREAD

**Issue by MobileMon** [atljeremy/JFMinimalNotifications#38]
The view won't dealloc unless I manually remove it from superView in the didDismissDelegateMethod

The problem with this though is if minimalNotificationDidDismissNotification method is never called, then the view is never dismissed. It may never get called if the view controller that called it is no longer alive, like popping back from a detail view controller

You can actually still see the view when rotating portrait to landscape on iphone x

- (void)minimalNotificationDidDismissNotification:(JFMinimalNotification*)notification {
    NSLog(@"didDismissNotification");
    if (notification != nil){
        [notification removeFromSuperview];
    }
}

**Comment by atljeremy** [comment]
Well that's not good. I'll dig into this issue tonight and get something pushed up to address it. In the meantime, could you put together a simple project that demonstrates the issue and share it with me? That'll save some time in trying to reproduce the issue. Thanks!

**Comment by MobileMon** [comment]
I forked the project to demonstrate the issue. Use the example project:
https://github.com/MobileMon/JFMinimalNotifications

In JFMinimalNotification.m I added a NSLog statement when dealloc is called

Now push and pop from both buttons and notice dealloc is not called when I don't remove from superview

Also goto either detail screen and let the notification appear. Now rotate the device landscape and see that the notification is actually still there (its just out of view) meaning it never actually removed itself

**Comment by atljeremy** [comment]
FYI: I've converted this library to Swift and have tested the rotation issue and I'm not seeing the notification during rotation. Regarding removing the notification from the superview, that is up to you to handle when you want to remove it. The notification itself doesn't remove itself from the superview when calling `dismiss`, it simply animates the layout constraints to hide it outside of the visual frame area. 

Also, I've tested that the delegate is being called for all visibility callback events. This includes `will..` and `did...` for both showing and hiding the notification.
